pytorch_lightning/utilities/data.py

```diff
@@ -329,7 +329,13 @@ def wrapper(obj: DataLoader, *args: Any, **kwargs: Any) -> None:
         params = dict(inspect.signature(obj.__init__).parameters)
         params.pop("args", None)
         params.pop("kwargs", None)
+        cls = type(obj)
         for arg_name, arg_value in chain(zip(params, args), kwargs.items()):
+            if hasattr(cls, arg_name) and getattr(cls, arg_name).fset is None:
+                # the class defines a read-only (no setter) property of this name. it's likely that the implementation
+                # will set `self._arg_name = arg_value` in `__init__` which is the attribute returned by the `arg_name`
+                # property so we are fine skipping in that case
```

Comment:
you know you got really confused about a bug when you write a 3 line comment for a 2 line change